pmkoalas/models/petrinet.py

```diff
@@ -0,0 +1,341 @@
+                   self._places == other._places and \
+                   self._transitions == other._transitions and \
+                   self._arcs   == self._arcs
+        return False
+
+def verbosecmp(obj1,obj2):
+    if obj1 == obj2:
+        return "Same"
+    result = ""
+    if obj1.__dict__.keys() != obj2.__dict__.keys():
+        ks1 = set(obj1.__dict__.keys())
+        ks2 = set(obj2.__dict__.keys())
+        result += f"Attributes not in both objects: {ks1 ^ ks2}\n"
+    if obj1.__dict__ != obj2.__dict__:
+        for k in obj1.__dict__:
+            if k in obj2.__dict__ and obj1.__dict__[k] != obj2.__dict__[k] :
+                result += f"Attribute {k} differs:\n"\
+                        + f"  {obj1.__dict__[k]} vs\n  {obj2.__dict__[k]}\n"
+    return result
+
+
+
+class Place:
+    def __init__(self,name,pid=None):
+        self._name = name
+        self._pid = pid
+
+    @property 
+    def name(self):
+        return self._name
+
+    @property
+    def pid(self):
+        return self._pid
+
+    @property
+    def nodeId(self):
+        return self._pid
+
+    def __eq__(self,other):
+        if type(other) == type(self):
+            return self.name == other.name and self.pid == other.pid
+        return False
+
+    def __hash__(self):
+        return hash((self._name,self._pid))
+
+    def __str__(self):
+        return f'({self.name})'  if self._pid is None \
+               else f'({self.name}({self._pid}))'
+
+    def __repr__(self):
+        return 'Place:' + str(self)
+
+
+class Transition:
+    def __init__(self,name,tid=None,weight=None,silent=None):
+        self._name = name
+        self._tid = tid
+        if weight is None:
+            self._weight = 1
+        else:
+            self._weight = weight
+        if not silent:
+            self._silent = False
+        else:
+            self._silent = True
+
+    @property 
+    def name(self):
+        return self._name
+
+    @property
+    def silent(self):
+        return self._silent
+
+    @property
+    def tid(self):
+        return self._tid
+
+    @property
+    def nodeId(self):
+        return self._tid
+
+    @property
+    def weight(self):
+        return self._weight
+
+    @weight.setter
+    def weight(self,value):
+        self._weight = value
+
+    def __eq__(self,other):
+        if type(self) == type(other):
+            return self.name == other.name and self.tid == other.tid and  \
+                   self.weight == other.weight and self._silent == other._silent
+        return False
+
+    def __hash__(self):
+        return hash((self._name,self._tid,self._weight,self._silent))
+
+    def __str__(self):
+        return f'[{self.name} {self.weight}]'  if self._tid is None \
+               else f'[{self.name}({self._tid}) {self.weight}]'
+
+    def __repr__(self):
+        return 'Transition:' + str(self)
+
+
+SILENT_TRANSITION_DEFAULT_NAME='tau'
+def silentTransition(name=None,tid=None,weight=None):
```

Comment:
I don't see how this would be helpful for a data structure. Can you tell me some more about this function?

Reply:
You create a silent transition by calling this function, and you don't need to name it if you don't want to.

Reply:
Eg, from state snapshot:
```
        tranId += 1
        tran = silentTransition(tid = tranId)
        atot[(fromPlaces,toPlaces)] = tran
        tweights[tran] = 1
```